pkg/cmd/add/add_app_test.go

```diff
@@ -80,6 +80,11 @@ func TestAddAppForGitOps(t *testing.T) {
 			DevEnv:     testOptions.DevEnv,
 			HelmUpdate: true, // Flag default when run on CLI
 		}
+		envDir, err := o.CommonOptions.EnvironmentsDir()
```

Comment:
is this test retrying!?

Reply:
Because the tests want to look at the contents of the resulting PR to make sure it did the right thing, and because of how `FakeProvider` works, that means we've gotta keep around the repo we create the PR in. Hence the `CloneDir` option I've added to all the various commands but not exposed via the CLI - a way to say "actually, don't use a temporary directory, use this one".

Reply:
So we need to add this just for testing purposes? Seems wrong, especially since it confuses the reader of the production code. I already stumbled across your comment.

Is there no other way to test this?

Reply:
Not so far as I can tell - opening a PR with `FakeProvider` just adds a record on the target repository's struct saying "there's a PR from this other repo", so once the other repo's directory gets deleted, the actual content of the PR is gone. So we either need to do the `CloneDir` stuff or we need to remove checking the content of the PR from the tests, which seems like a bad idea.

Reply:
Another approach would be to work against known locations. As I mentioned before, I think we should create all these temporary directories in a common base, aka JX_HOME/tmp. This way we have a place to look/scan for the created directory. 

This combined with creating the tmp directory quite high up in the call hierachy (somewhere close to _Run_), would probably get us closter to where we want to be.

For now that is probably too much to ask for ;-)

Reply:
> So we either need to do the CloneDir stuff or we need to remove checking the content of the PR from the tests, which seems like a bad idea.

Maybe the tests could be split as well. Personally, I would be very reluctant to have this flag just because I need it for testing. For me it is a code smell.